server/src/main/java/org/elasticsearch/search/fetch/FetchSubPhase.java

```diff
@@ -39,13 +40,15 @@
         private IndexSearcher searcher;
         private LeafReaderContext readerContext;
         private int docId;
+        private final SourceLookup sourceLookup = new SourceLookup();
```

Comment:
I think we should add a comment here about why we don't want to share the lookup in QueryShardContext.

Reply:
👍

Reply:
I added Javadoc on `HitContext#sourceLookup` and `FetchPhase#prepareHitContext` to explain the logic. Let me know if it could use more clarification.